pkg/lockfile/diagnostics.go

```diff
@@ -0,0 +1,24 @@
+package lockfile
+
+import (
+	"fmt"
+	"os"
+)
+
+type Diagnostics struct {
+	Warnings []string
+}
+
+func (diag *Diagnostics) Warn(warn string) {
+	diag.Warnings = append(diag.Warnings, warn)
+}
```

Comment:
Assuming there are ever cases where the warnings are duplicated, the consumer will need to consider de-duplication to present them to users. Not sure if it's worth it, but maybe a `map[string]struct{}` or something like that could provide that out the box? What do you think?

Reply:
I feel at this point it's best to leave that to the consumer to handle as there's no defined spec around these warnings and _hopefully_ we shouldn't need to expand this further, but also if we do it should be easy to add a `WarningsMap` or something.